onnx/defs/nn/defs.cc

```diff
@@ -2487,4 +2487,195 @@ ONNX_OPERATOR_SET_SCHEMA(
+      final_image_shape->add_dim();
+    }
+
+    // image_shape is appended to the end
+    int n_image_shape = ctx.getInputType(1)->tensor_type().shape().dim(0).dim_value();
+    for (int i=0; i < n_image_shape; ++i) {
+      final_image_shape->add_dim();
+    }
+  }
+
+  return;
+}
+
+static const char* Col2Im_ver16_doc = R"DOC(
+The operator rearranges the a [H, W] or [N, H, W] input matrix columns into blocks.
+
+Col2Im behaves like PyTorch's fold for two output spatial dimensions https://pytorch.org/docs/stable/generated/torch.nn.Fold.html
+
+NOTE: Only 4-D output tensors (batched image-like tensors) are supported.
+
+NOTE: Although specifying image_shape looks redundant because it could be calculated from
+      convolution formulas, it is required as input for more advanced scenarios as explained
+      at PyTorch's implementation (https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/Col2Im.cpp#L10)
+
+)DOC";
+
+ONNX_OPERATOR_SET_SCHEMA(
+    Col2Im,
+    16,
+    OpSchema()
+    .Attr(
+        "dilations",
+        "dilation value along each spatial axis of the image. "
+        "If not present, the dilation defaults is 1 along each spatial axis of the image.",
+        AttributeProto::INTS,
+        OPTIONAL_VALUE)
+	  .Attr(
+        "pads",
+        "Padding for the beginning and ending along each spatial axis, it can take any value greater "
+        "than or equal to 0. The value represent the number of pixels added to the beginning "
+        "and end part of the corresponding axis. `pads` format should be as follow "
+        "[x1_begin, x2_begin...x1_end, x2_end,...], where xi_begin the number of pixels "
+        "added at the beginning of axis `i` and xi_end, the number of pixels added at "
+        "the end of axis `i`. If not present, the padding defaults to 0 along start and end of each spatial axis.",
+        AttributeProto::INTS,
+        OPTIONAL_VALUE)
+    .Attr(
+        "strides",
+        "Stride along each spatial axis. "
+        "If not present, the stride defaults to 1 along each spatial axis.",
+        AttributeProto::INTS,
+        OPTIONAL_VALUE)
+    .SetDoc(Col2Im_ver16_doc)
+    .Input(
+        0,
+        "input",
+        "Input data tensor to be rearranged from column blocks back into an image."
+        "`input` format should be either [H, W] or [N, H, W].",
+        "T",
+        OpSchema::Single,
+        true,
+        1,
+        OpSchema::Differentiable)
+    .Input(
+        1,
+        "image_shape",
+        "The shape of the image after rearranging the column blocks."
+        "`image_shape` format should be a 2-tuple as follow [H, W].",
+        "tensor(int64)",
+        OpSchema::Single,
+        true,
+        1,
+        OpSchema::NonDifferentiable)
+    .Input(
+        2,
+        "block_shape",
+        "The shape of the block to apply on the input."
+        "`block_shape` format should be a 2-tuple as follow [H, W].",
```

Comment:
Suggest changing this as below. I am using KH/KW for kernel-height and kernel-width; not sure what is commonly used "block" or "kernel"?
The shape of the block/kernel. This is a 1-dimensional tensor of size 2, containing the value (KH, KW).